genevents.go

```diff
@@ -0,0 +1,367 @@
+// Copyright 2019 The Ebiten Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+)
+
+type Event struct {
+	Comment string
+	Members []Member
+}
+
+func (e *Event) Name() string {
+	return strings.Split(e.Comment, " ")[0]
+}
+
+type Member struct {
+	Comment string
+	Type    string
+}
+
+func (m *Member) Name() string {
+	return strings.Split(m.Comment, " ")[0]
```

Comment:
`return strings.SplitN(m.Comment, " ", 2)[0]`
will be more efficient.

Reply:
Good point. Done.